traffic-manager-application-gateway-demo-setup/azuredeploy.json

```diff
@@ -96,43 +96,20 @@
     "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
     "addressPrefix": "10.0.0.0/16",
     "subnetPrefix": "10.0.0.0/28",
-    "nestedTemplatesFolder":  "nested",
-    "appGwTemplateName": "azuredeployappgw.json"
+    "nestedTemplatesFolder": "nested",
+    "appGwTemplateName": "azuredeployappgw.json",
+    "location1": "[parameters('primarylocation')]",
+    "location2": "[parameters('secondarylocation')]"
```

Comment:
Maybe use: 
    "location": [
      "[parameters('primaryLocation')]", 
      "[parameters('secondaryLocation')]"
    ]
And then reference via:
variables('location')[copyIndex()]